### DISCUSSION TITLE
Custom own Record ID from client side

### DISCUSSION THREAD

**Discussion by iamdineshkumar** [pocketbase/pocketbase#3173] [Q&A]
I am planning about create an application in flutter with offline compatibility. So ID need to be generated in offline db. To avoid duplication, I thought about using BSON Object ID(https://github.com/mongodb/specifications/blob/master/source/objectid.rst) and its hex value will be 24 character. But at the moment Pocketbase restrict id as 15 character only. There is no in-build function provided in sdk to generate id. While checking backend code its random alphanumeric string with 15 character length precise. So whether its possible to remove the restriction while using framework? or How to handle this?

**Comment by ganigeorgiev** [comment]
> So whether its possible to remove the restriction while using framework? or How to handle this?

No at the moment. Support for custom ids format and length may be added in the future as there plans to allow more "native" SQLite dev experience, but for now this remains a low priority.

**Reply by iamdineshkumar** [reply]
Noted. At the moment, I have wrote my own logic to generate time based random alphanumeric id with 15 character precision as work around.
